v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -28,63 +29,31 @@ import { getHardhatRuntimeEnvironmentSingleton } from "../hre-singleton.js";
 
 export async function main(cliArguments: string[]) {
   const hreInitStart = performance.now();
-  let configPath: string | undefined;
-  let showStackTraces: boolean = false; // true if ci
-  let help: boolean = false;
-  let version: boolean = false;
 
   const usedCliArguments = new Array(cliArguments.length).fill(false);
 
-  for (let i = 0; i < cliArguments.length; i++) {
-    const arg = cliArguments[i];
-
-    if (arg === "--config") {
-      usedCliArguments[i] = true;
-
-      if (configPath !== undefined) {
-        throw new Error("Multiple --config arguments are not allowed");
-      }
-
-      configPath = cliArguments[i + 1];
-      i++;
-
-      usedCliArguments[i] = true;
-      continue;
-    }
-
-    if (arg === "--show-stack-traces") {
-      usedCliArguments[i] = true;
-      showStackTraces = true;
-      continue;
-    }
-
-    if (arg === "--help") {
-      usedCliArguments[i] = true;
-      help = true;
-      continue;
-    }
-
-    if (arg === "--version") {
-      usedCliArguments[i] = true;
-      version = true;
-      continue;
-    }
-  }
+  const hhGlobalParams = await parseInitialHardhatParameters(
```

Comment:
The function and variable names are a bit confusing, as we have another function with a similar name (`parseGlobalArguments`). What do you think about `parseSpecialCliArguments`? For the variable, I'd consider destructuring it, but another option could be `hardhatSpecialArgs`.

Reply:
Agreed. I mostly find it confusing that the names are not consistent. I'm ok with "initial", but here they are both all initial and global, and there's another global.